### PR TITLE
13 send user input using gpt api and output received text

### LINKED ISSUE #13: Send user input using GPT api and output received text
## 어떤 기능인가요?

> GPT api 연결하고 User input 받아서 보내고 response text 출력

## 작업 상세 내용

- [x] api 키 생성
- [x] api 연결
- [ ] 응답 출력
- [ ] 채팅형식 구현

### PR DESCRIPTION
**Pull request by dan05s**
## #️⃣연관된 이슈

> #13 
## 📝작업 내용

> VSCode extension activity bar에 webview 구현. webview에 사용자 input send 버튼을 통해 받고 gpt api 활용하여 응답받고 webview에 채팅 형식으로 출력. reset버튼 클릭 시 기록 삭제

## 💬리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>

### CODE REVIEW

File: CHANGELOG.md
Comment (young170) [inline CHANGELOG.md:1]: Keep our project name, coding-roadmap-assistant

File: README.md
Comment (young170) [inline README.md:1]: Keep our project name, coding-roadmap-assistant

File: media/icon.webp
Comment (young170) [inline media/icon.webp:1]: Where is this used? Need to delete if not used

File: media/webview.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="ko">
+  <head>
+    <meta charset="UTF-8" />
+    <title>Webview</title>
```
Comment (young170) [inline media/webview.html]: Update `<title>` to match project name, CRA (Coding Roadmap Assistant)

File: media/webview.js
Comment (young170) [inline media/webview.js:1]: Why .js?

File: media/webview.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <div id="chat-container">
+      <ul id="list"></ul>
+      <!-- 대화 내용이 올라가는 영역 -->
+      <div id="input-area">
+        <input type="text" placeholder="여기에 입력하세요" id="input" />
+        <button id="send">Send</button>
+        <button id="reset">Reset</button>
+      </div>
+    </div>
+    <script src="webview.js"></script>
```
Comment (young170) [inline media/webview.html]: Change .js to .ts

File: package-lock.json
Comment (young170) [inline package-lock.json:1]: Keep our project name, coding-roadmap-assistant

File: package.json
Comment (young170) [inline package.json:1]: Keep our project name, coding-roadmap-assistant

File: package.json [hunk 3]
```diff
@@ -43,10 +63,12 @@
     "@vscode/test-electron": "^2.4.1",
     "esbuild": "^0.24.0",
     "eslint": "^9.16.0",
-    "eslint-config-prettier": "^9.1.0",
```
Comment (young170) [inline package.json]: Why are eslint and prettier dependencies removed?

File: package.json [hunk 4]
```diff
@@ -43,10 +63,12 @@
...
-    "eslint-plugin-prettier": "^5.2.1",
     "npm-run-all": "^4.1.5",
-    "prettier": "^3.4.2",
     "typescript": "^5.7.2"
+  },
+  "dependencies": {
+    "axios": "^1.7.9",
```
Comment (young170) [inline package.json]: What and why is axios used?

File: package.json [hunk 5]
```diff
@@ -43,10 +63,12 @@
...
+    "dotenv": "^16.4.7",
```
Comment (young170) [inline package.json]: What and why is dotenv used?

File: src/webview.ts
Comment (young170) [inline src/webview.ts:1]: Don't forget to update all .js instances to .ts

**Review by young170 — COMMENTED** [review]
Please address comments then ask for new review.

**Review by young170 — CHANGES_REQUESTED** [review]
Please  address comments

File: package.json (hunk 4 above)
Comment (Jangmyun) [reply package.json]: eslint-prettier plugin is needed

[ky](https://www.npmjs.com/package/ky) 쓰는 게 axios보다 익스텐션 사용목적에 부합할 것으로 보임.

File: src/webview.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        localResourceRoots: [
+          vscode.Uri.joinPath(context.extensionUri, "media"), // media 폴더를 로컬 리소스로 설정
+        ],
+      };
+
+      // HTML 파일 읽기
+      const styleUri = webviewView.webview.asWebviewUri(
+        vscode.Uri.joinPath(context.extensionUri, "media", "styles.css")
+      );
+      const htmlPath = path.join(
```
Comment (Jangmyun) [inline src/webview.ts]: 이런 간단한 로직은 **path**를 사용하지 않아도 쉽게 구현가능함 (string concatenating)

**Review by Jangmyun — CHANGES_REQUESTED** [review]

**Review by young170 — COMMENTED** [review]
Seems like all comments were applied. Waiting for @Jangmyun to approve :)

**Pull request merged**